### DISCUSSION TITLE
How does the Static prerendering handle POST endpoints?

### DISCUSSION THREAD

**Discussion by HenrijsS** [sveltejs/kit#6737] [Q&A]
This may seem a bit out there, but I was wondering: How does Static Prerendering handle POST endpoints (+page.server.js) files on a CDN?

I want to run a Sendgrid function with Environment Variables by [submitting a form ](https://kit.svelte.dev/docs/form-actions) and I'm worried that it might expose the environment variables.

The site will be hosted on CloudFlare pages. Do endpoints invoke workers or do they function differently?

**Comment by teemingc** [comment]
I believe if you use the [private env modules](https://kit.svelte.dev/docs/modules#$env-dynamic-private), it will be difficult to accidentally expose environment variables.

~Also, I don't think prerendering will have any affect on your form `actions`. Prerendering runs your `load` functions at build time and uses the values returned to build a `html` file for your page.~

**Comment by kesdigital** [comment]
Does prerendering work on pages with actions? , I get this error whenever I try to prerender a page with actions.

```bash
Cannot prerender pages with actions
Error: Cannot prerender pages with actions
```

```ts
// page.server.ts

export const actions: Actions = {
  default: async ({ request }) => {
    const formData = await request.formData();

    const email = formData.get('email');
    const firstName = formData.get('firstName');
    const lastName = formData.get('lastName');
    const phoneNumber = formData.get('phoneNumber');

    if (email && firstName && lastName && phoneNumber) {
      return { success: true }
    }

    return { failure: true }

  }
}

export const prerender = true;
```

**Reply by HenrijsS** [reply]
I just deployed my site on CloudFlare pages with `adapter-auto` and didn't specify the `prerender = true`  attribute anywhere.
It all applies automatically and I still have no clue how it published endpoints.

Try removing the prerender. Also, where do you plan to host it?

**Reply by kesdigital** [reply]
Am using `prerender = true` in my root layout, but I get the error and am forced to `prerender = false` on any page with actions
And I plan to host it on Vercel. Using `adapter-auto` as well.
